### PR TITLE
EntryPoint Outbox updates and dispatch integration event shell

### PR DESCRIPTION
**Pull request by Korgath**
## Description

- SimpleInjectorMediatrContainerExtension updated
- Simple Injector implemented in EntryPoint Outbox
- Outbox Manager implemented
- EventMessageDispatcher function added
- Shell implementation of integration for event dispatching from Outbox
- ConsumerRegistered doman event and integration event files added

## References

* #561

### CODE REVIEW

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.EntryPoints.Common/MediatR/SimpleInjectorMediatorContainerExtensions.cs [hunk 1]
```diff
@@ -56,15 +44,13 @@ public static Container BuildMediatorWithPipeline(this Container container, Asse
                 typeof(RequestPostProcessorBehavior<,>),
             };
 
-            // Register pipeline
+            // Register built both-in and custom pipeline
```
Comment (cmgrd) [inline source/business-workflow/source/Energinet.DataHub.MarketRoles.EntryPoints.Common/MediatR/SimpleInjectorMediatorContainerExtensions.cs:47]: nit: typo, should be "both built-in..." I think

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.EntryPoints.Outbox/RequestHandlers/ConsumerRegisteredDispatch.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+// limitations under the License.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Energinet.DataHub.MarketRoles.Infrastructure.Integration.IntegrationEventDispatching.MoveIn;
+using MediatR;
+
+namespace Energinet.DataHub.MarketRoles.EntryPoints.Outbox.RequestHandlers
+{
+    public class ConsumerRegisteredDispatch : IRequestHandler<ConsumerRegisteredIntegrationEvent>
```
Comment (cmgrd) [inline source/business-workflow/source/Energinet.DataHub.MarketRoles.EntryPoints.Outbox/RequestHandlers/ConsumerRegisteredDispatch.cs:22]: Dispatcher? Or Handler?

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Infrastructure/Integration/IntegrationEventDispatching/Channels/ServiceBusChannel.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Energinet.DataHub.MarketRoles.Infrastructure.Transport;
+
+namespace Energinet.DataHub.MarketRoles.Infrastructure.Integration.IntegrationEventDispatching.Channels
+{
+    public class ServiceBusChannel : Channel
```
Comment (cmgrd) [inline source/business-workflow/source/Energinet.DataHub.MarketRoles.Infrastructure/Integration/IntegrationEventDispatching/Channels/ServiceBusChannel.cs:21]: We will probably have more than one service bus implementation, should we add something specific to the name?

**Review by cmgrd — DISMISSED** [review]

**Review by kft — CHANGES_REQUESTED** [review]
Currently we are not supposed to publish an integration event on the `ConsumerRegistered´ event. We should just publish the `MoveInRequestApproved` actor message.
We will create an integration event when we effectuate the move in request.

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Infrastructure/Integration/Contracts/MoveIn/MoveInIntegrationEventContract.proto [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+syntax = "proto3";
+
+option csharp_namespace = "Energinet.DataHub.MarketRoles.IntegrationEventContracts";
+
+message ConsumerRegisteredIntegrationEvent {
+  string MeteringPointId = 1;
+  bool CustomerMovedIn = 2;
+}
+
+message MoveInIntegrationEventEnvelope {
```
Comment (cmgrd) [inline source/business-workflow/source/Energinet.DataHub.MarketRoles.Infrastructure/Integration/Contracts/MoveIn/MoveInIntegrationEventContract.proto:25]: Do we know how/when to "group" integration events?

**Review by cmgrd — APPROVED** [review]

**Review by MrDach — APPROVED** [review]

**Pull request merged**